frontend/src/components/FilterDrawer.vue

```diff
@@ -222,7 +224,13 @@ const checkSearchTerm = (searchTerm, fieldsToSearch, feature) => {
 
 async function filter() {
   emit('onFilter', { selectedSpatialZones, selectedTemporalZones, selectedProcesses })
-
+  if (typeof window !== 'undefined' && window.heap) {
```

Comment:
because we call the `filter` function on several events [here](https://github.com/CUAHSI/hydroprocess-db/blob/logging_metrics/frontend/src/components/FilterDrawer.vue#L78-L80):
```
          @update:focused="filter"
          @keydown.enter.prevent="filter"
          @click:clear="filter"
```
This is going to send a null value for `textSearched` to Heap every time we click/blur the text search `v-text-field`

Reply:
https://github.com/user-attachments/assets/19446e39-d905-4466-8f1c-58aacf07d751

Reply:
also because we call the `filter` function from other filter components, we will be sending Heap search data every time that we change a filter. This would log a bunch of "search" events in Heap, even though the user was actually just adding/removing model filters

we call it:
https://github.com/CUAHSI/hydroprocess-db/blob/logging_metrics/frontend/src/components/FilterDrawer.vue#L42
https://github.com/CUAHSI/hydroprocess-db/blob/logging_metrics/frontend/src/components/FilterDrawer.vue#L54
https://github.com/CUAHSI/hydroprocess-db/blob/logging_metrics/frontend/src/components/FilterDrawer.vue#L65

https://github.com/user-attachments/assets/951d0e3e-c6b8-448b-9e06-f97895e3bc99